### ISSUE TITLE
Should topology health check be part of the run method?

### ISSUE THREAD

**Issue by ma-sadeghi** [PMEAL/OpenPNM#2622]
I've noticed that a significant portion of `Transport`'s run time (somewhere between 40% to 70% depending on network size) is spent in `_validate_topology_health`. Do we really need it?

**Comment by jgostick** [comment]
We'll this is basically the same as checking to make sure the coefficient matrix won't produce a singular matrix.  But perhaps we should add a ``check_algorithm_health`` function which users can call on demand instead of making it mandatory.

**Comment by ma-sadeghi** [comment]
Yeah, making it a public method which users can call on demand sounds like a good idea.

To back up, I think if the matrix is ill-conditioned, the matrix solver fails to converge, especially the direct ones, so it's kind of already covered by the solver. Iterative solvers are different though, but if a user is advanced enough to pass their own solver to the run method, they might as well learn how to do diagnostics (i.e., call `check_algorithm_health`, etc.).

On the other hand, these are all performance optimizations the only become relevant when the network becomes large enough. You could argue that for most use cases, it's not noticeable.
